lib/Mojolicious/Plugin/CGI.pm

```diff
@@ -215,10 +215,14 @@ sub register {
         $| = 1;
         select STDOUT;
         $| = 1;
-        { exec $self->{script} }
-        die "Could not execute $self->{script}: $!";
+	if ($self->{run}) {
+	  $self->{run}->();
```

Comment:
I guess I was just confused by this line of yours: `$c->$code($self);` – how would `$c->$code` ever work, if `$code` is something like `\&OddMuse::DoWikiRequest` as it is in my case? I do agree, though, that passing the controller on to the code ref would allow the code ref to do other interesting stuff. So that would be:
```
        if (my $code = $self->{run}) {
          Mojo::IOLoop->reset;    # clean up
          $code->($c);
          exit;
        }
```
I've added to a7f4bf6.